core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1962,6 +1962,21 @@ private[spark] object Utils extends Logging {
     path
   }
 
+  /**
+   * Updates Spark config with properties from a set of Properties.
+   * Provided properties have the highest priority.
+   */
+  def updateSparkConfigFromProperties(
+      conf: SparkConf,
+      properties: Map[String, String]) : Unit = {
+    properties.filter { case (k, v) =>
+      k.startsWith("spark.")
+    }.foreach { case (k, v) =>
+      conf.set(k, v)
+      sys.props.getOrElseUpdate(k, v)
```

Comment:
loadDefaultSparkProperties method a few lines above does that so I thought this is a common practice to load stuff there because they might be reused elsewhere. I can remove it for sure.